tools/config-generator/main_test.go

```diff
@@ -43,3 +54,869 @@ func TestOutputConfig(t *testing.T) {
 		t.Fatalf("Output count should have been exactly 1, but was %d", output.count)
 	}
 }
+
+func TestReadTemplate(t *testing.T) {
+	SetupForTesting()
+	templatesCache["foo"] = "bar"
+	if diff := cmp.Diff(readTemplate("foo"), "bar"); diff != "" {
+		t.Errorf("Cached template was not returned: (-got +want)\n%s", diff)
+	}
+
+	readTemplate("non/existent/file/path")
+	if logFatalCalls != 1 {
+		t.Errorf("Non existent file should have caused error")
+	}
+
+	delete(templatesCache, "foo")
+}
+
+func TestNewbaseProwJobTemplateData(t *testing.T) {
+	SetupForTesting()
+	out := newbaseProwJobTemplateData("foo/subrepo")
+	if diff := cmp.Diff(out.PathAlias, ""); diff != "" {
+		t.Errorf("Unexpected path alias: (-got +want)\n%s", diff)
+	}
+
+	pathAliasOrgs.Insert("foo")
+	out = newbaseProwJobTemplateData("foo/subrepo")
+	expected := "path_alias: knative.dev/subrepo"
+	if diff := cmp.Diff(out.PathAlias, expected); diff != "" {
+		t.Errorf("Unexpected path alias: (-got +want)\n%s", diff)
+	}
+
+	nonPathAliasRepos.Insert("foo/subrepo")
+	out = newbaseProwJobTemplateData("foo/subrepo")
+	if diff := cmp.Diff(out.PathAlias, ""); diff != "" {
+		t.Errorf("Unexpected path alias: (-got +want)\n%s", diff)
+	}
+
+	// don't pollute the global setup
+	pathAliasOrgs.Delete("foo")
+	nonPathAliasRepos.Delete("foo/subrepo")
+}
+
+func TestCreateCommand(t *testing.T) {
+	SetupForTesting()
+	preCommand = "" // global
+	in := baseProwJobTemplateData{Command: "foo", Args: []string{"bar", "baz"}}
+	out := createCommand(in)
+	expected := "[foo bar baz]"
+	if diff := cmp.Diff(fmt.Sprintf("%v", out), expected); diff != "" {
```

Comment:
`cmp.Diff` can compare slice too, can do you
```
expected := []string{"foo", "bar", "baz"}
if diff := cmp.Diff(expected, out)
```
instead?

Reply:
There are a few other cases like this down below as well

Reply:
Done